reference/pipeline/expressions/index.md

```diff
@@ -0,0 +1,207 @@
+`.toString()` on the result. For example:
+`${#stage("Deploy")["status"].toString() == "SUCCEEDED"}`.
+
+### Functions
+
+The expression language has several built in helper functions that simplify
+common use cases. The syntax for calling built-in functions is
+`#functionName(params)`. For example, `#fromUrl("http://www.netflix.com")`. See
+the full list of functions [here](#helper-functions).
+
+### Lists
+
+You can also use lists in your expressions. Spinnaker always provides a list
+called _stages_, which you can use to access each stage by index:
+`${execution["stages"][0]}` returns the value of the first stage in your
+pipeline. Keep in mind, though, that if you change the order in which your
+stages execute in the pipeline, the value of `stages[i]` will also change.
+
+### Maps
+
+You can use maps to access data from the JSON representation of your pipeline.
+For example, `${trigger["properties"]["example"]}` evaluates to the value of
+the `example` trigger property. To see the available values for a given
+pipeline, view the execution JSON:
+1. Go to an execution of your pipeline.
+2. Click on _Details_ to expand the pipeline.
+3. Click on the source link under the pipeline execution, which opens a new tab
+    containing the JSON details of your pipeline execution.
+    _[[screenshot](https://screenshot.googleplex.com/U1rVRjdiqKC.png)]_
+
+Note that while Spinnaker supports both dot (`map.value`) and square bracket
+(`map["value"]`) notation, we recommend using square brackets. That is, prefer
+`trigger["properties"]["value"]` instead of `trigger.properties.value`. There
+are a few places where using dot notation produces unexpected results, such as
+after a filter operation or when getting nested JSON values from an URL.
+
+You can filter maps using `.?`. For example, to filter a list of stages by type,
+you would use `${execution["stages"].?[type == "bake"]}`.
+
+### Math
+
+You can use arithmetic operations in your expressions, such as
+`${trigger["buildInfo"]["number"] * 2}`.
+
+### Strings
+
+Strings evaluate to themselves: `${"This is a String."}` becomes "This is a
+String."
+
+## Helper properties
+
+[Helper properties](/guides/user/pipeline/expressions/#helper-properties) are
+attribute shortcuts that you can use in Spinnaker. The specific properties are:
+
+* `execution`: refers to the current pipeline execution.
+* `parameters`: refers to pipeline parameters.
+* `trigger`: refers to the pipeline trigger.
+* `scmInfo`: refers to the git details of either the trigger or the most
+recently executed Jenkins stage.
+  * `scmInfo.sha1` returns the hash of the last build
+  * `scmInfo.branch` returns the branch name
+* `deployedServerGroups`: refers to the server group that was created by the
+  last deploy stage. It should look something like:
+  `{"account":"my-gce-account",
+    "capacity": { "desired":1.0,"max":1.0,"min":1.0},
+    "region":"us-central1",
+    "serverGroup":"gcespin-dev-v005"}`
+
+## Helper functions
```

Comment:
This list occurs in /guides/ too. Recommend changing that so all that's in /guides/ is a sentence describing what help functions to, generally, and linking to this reference.

Reply:
Worth mentioning that `deployedServerGroups` can also be used as a function: `${#deployedServerGroups("Deploy in us-east-1")}`. Takes the name of the deploy stage.